lib/msal-react/src/MsalProvider.tsx

```diff
@@ -47,6 +47,8 @@ export function MsalProvider({instance, children}: MsalProviderProps) {
             }
         });
 
+        instance.handleRedirectPromise();
```

Comment:
Why is this needed here?

Reply:
I see the description now. What if an app needs to wait until `handleRedirectPromise` is completed in order to render? How would that work (given there is no longer a hook they can use)?

Reply:
I guess it depends what their render condition is. If the app needs to render based on whether or not a user is signed in it can use the `useIsAuthenticated` hook which would update after `handleRedirectPromise` finishes. 

If a component needs a token and needs to ensure `handleRedirectPromise` is done before calling `acquireTokenSilent` we will probably need to add an `interactionInProgress` boolean alongside the existing `loginInProgress` and we might need to move up the handleRedirect Start event.